src/telliot_feeds/reporters/reporter360.py

```diff
@@ -0,0 +1,181 @@
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class StakerInfo:
+    """Data types for staker info
+    start_date: TimeStamp
+    stake_balance: int
+    locked_balance: int
+    reward_debt: int
+    last_report: TimeStamp
+    reports_count: int
+    gov_vote_count: int
+    vote_count: int
+    in_total_stakers: bool
+    """
+
+    start_date: TimeStamp
+    stake_balance: int
+    locked_balance: int
+    reward_debt: int
+    last_report: TimeStamp
+    reports_count: int
+    gov_vote_count: int
+    vote_count: int
+    in_total_stakers: bool
+
+
+class reporter360(TellorFlexReporter):
+    def __init__(
+        self,
+        endpoint: RPCEndpoint,
+        account: ChainedAccount,
+        chain_id: int,
+        oracle: Contract,
+        token: Contract,
+        autopay: Contract,
+        stake: Optional[int] = None,
+        datafeed: Optional[DataFeed[Any]] = None,
+        expected_profit: Union[str, float] = "YOLO",
+        transaction_type: int = 2,
+        gas_limit: int = 350000,
+        max_fee: Optional[int] = None,
+        priority_fee: int = 100,
+        legacy_gas_price: Optional[int] = None,
+        gas_price_speed: str = "safeLow",
+        wait_period: int = 7,
+    ) -> None:
+
+        self.endpoint = endpoint
+        self.oracle = oracle
+        self.token = token
+        self.autopay = autopay
+        self.stake: Optional[int] = stake  # type: ignore
+        self.datafeed = datafeed
+        self.chain_id = chain_id
+        self.acct_addr = to_checksum_address(account.address)
+        self.last_submission_timestamp = 0
+        self.expected_profit = expected_profit
+        self.transaction_type = transaction_type
+        self.gas_limit = gas_limit
+        self.max_fee = max_fee
+        self.wait_period = wait_period
+        self.priority_fee = priority_fee
+        self.legacy_gas_price = legacy_gas_price
+        self.gas_price_speed = gas_price_speed
+        self.autopaytip = 0
+        self.staked_amount: Optional[float] = None
+        self.staker_info: Optional[StakerInfo] = None
+        self.allowed_stake_amount = 0
+
+    async def ensure_staked(self) -> Tuple[bool, ResponseStatus]:
+        # get oracle required stake amount
+        stake, status = await self.oracle.read("getStakedAmount")
+
+        if (not status.ok) or (stake is None):
+            msg = "Unable to read current stake amount"
+            return False, error_status(msg, log=logger.info)
+
+        # get accounts current stake total
+        staker_info, status = await self.oracle.read(
+            "getStakerInfo",
+            _stakerAddress=self.acct_addr,
+        )
+
+        if (not status.ok) or (staker_info is None):
+            msg = "Unable to read reporters staker info"
+            return False, error_status(msg, log=logger.info)
+
+        if self.stake is None:
+            self.stake = stake
+        else:
+            msg = "Stake amount has changed due to TRB price change."
```

Comment:
It'll set `self.stake` to some number the first time around, then the next time it'll always hit this condition, which would be wrong because the stake amount might not always change due to TRB price change?